pennylane/plugins/default_qubit.py

```diff
@@ -395,15 +454,22 @@ def mat_vec_product(self, mat, vec, wires):
         # We'll need to invert this permutation to put the indices in the correct place
         unused_idxs = [idx for idx in range(self.num_wires) if idx not in wires]
         perm = wires + unused_idxs
-        inv_perm = np.argsort(perm) # argsort gives inverse permutation
+        inv_perm = np.argsort(perm)  # argsort gives inverse permutation
         state_multi_index = np.transpose(tdot, inv_perm)
         return np.reshape(state_multi_index, 2 ** self.num_wires)
 
     def expval(self, observable, wires, par):
         if self.shots == 0:
             # exact expectation value
-            A = self._get_operator_matrix(observable, par)
-            ev = self.ev(A, wires)
+            if isinstance(observable, Tensor):
+                # unpack the observable matrix
+                # TODO: merge the matrix generator functions
+                A = self._get_tensor_operator_matrix(observable)
+                ev = self.ev(A, wires)
+            else:
+                A = self._get_operator_matrix(observable, par)
+                ev = self.ev(A, wires)
```

Comment:
This line is duplicated by both branches of the if statement, so can be moved below the `if-else`.